app/Main.hs

```diff
@@ -62,11 +62,37 @@ packageDir :: Text -> PackageName -> Text -> Turtle.FilePath
 packageDir set pkgName version =
   ".psc-package" </> fromText set </> fromText (runPackageName pkgName) </> fromText version
 
+-- a the source of a package set can be a git url or a local path
+data UrlOrLocal
+  = Url Text
+  | Local Text
+  deriving (Show, Generic)
+
+-- is the given text a url, as far as we care about sources?
+isUrl :: Text -> Bool
+isUrl txt =
+  let isHttp = T.isPrefixOf "https://" txt
+      isGit = T.isPrefixOf "git://" txt
+  in isHttp || isGit
+
+mkUrlOrLocal :: Text -> UrlOrLocal
+mkUrlOrLocal txt =
+  if isUrl txt
+    then Url txt
+    else Local txt
+
+instance Aeson.FromJSON UrlOrLocal where
+  parseJSON value = mkUrlOrLocal <$> Aeson.parseJSON value
```

Comment:
This will interpret http:// as a local path. It should probably fail and tell the user to use https:// instead.

Reply:
the url check is quite lazy, so i'm willing to change it with almost anything else. just need to know where to pull a better definition from

Reply:
in reality another lazy solution i guess would be to check if `://` is an infix, which accurately reflects the faces of pedants reading this comment

Reply:
third, i guess mots people will use `git@github.com:purescript/package-sets.git` as a git url, so the prefix is wrong in any case

Reply:
Alternatively you could require that local paths begin with file://. E.g. `file://../foo`.

Reply:
i hate file urls, especially considering that this should work with relative paths easily

Reply:
when i glance over docs i see that anything ending in .git is basically fair game for git https://git-scm.com/book/en/v2/Git-on-the-Server-The-Protocols